skimage/morphology/gray.py

```diff
@@ -171,6 +173,7 @@ def erosion(
     return out
 
 
+@dispatchable
 @default_footprint
 def dilation(
```

Comment:
In which order should the decorators be applied? dispatch first, then default footprint or the other way around?

Reply:
In this case I'd say outermost is the right call. `@default_footprint` could be considered an internal implementation detail. To users the parameter it touches is presented as `footprint=None`. 

We often use decorators for deprecations. If we dispatch in the outermost position, the backend would have to re-implement the deprecations. 

So it might make sense to use:

```python
@deprecations
@dispatching
@implementation_details  # e.g. input handling
def some_amazing_skimage_function():
    pass
```